src/main/java/com/gmail/woosay333/onlinebookstore/exception/handler/ControllerExceptionHandler.java

```diff
@@ -49,6 +54,21 @@ protected ResponseEntity<Object> handleHttpMessageNotReadable(
         return getResponseEntity(HttpStatus.valueOf(status.value()), ex.getLocalizedMessage());
     }
 
+    @ExceptionHandler(AccessDeniedException.class)
+    protected ResponseEntity<Object> handleAccessDenied(AccessDeniedException ex) {
+        return getResponseEntity(FORBIDDEN, ex.getMessage());
+    }
+
+    @ExceptionHandler(AuthenticationException.class)
+    protected ResponseEntity<Object> handleAuthenticationException(AuthenticationException ex) {
+        return getResponseEntity(UNAUTHORIZED, ex.getMessage());
+    }
+
+    @ExceptionHandler(UsernameNotFoundException.class)
+    protected ResponseEntity<Object> handleUsernameNotFoundException(UsernameNotFoundException ex) {
+        return getResponseEntity(NOT_FOUND, ex.getMessage());
+    }
+
```

Comment:
These exceptions handled automatically as far as I remember

Reply:
You're probably right. Found a topic about it on [stackoverflow](https://stackoverflow.com/questions/59417122/how-to-handle-usernamenotfoundexception-spring-security).
I'll fix it now and throw it in again for checking, thanks.